gateway-service/src/main/java/org/zowe/apiml/gateway/config/ConnectionsConfig.java

```diff
@@ -181,21 +177,31 @@ public HttpsFactory factory() {
     @Bean
     public NettyRoutingFilterApiml createNettyRoutingFilterApiml(HttpClient httpClient, ObjectProvider<List<HttpHeadersFilter>> headersFiltersProvider, HttpClientProperties properties) {
         // obtain SSL contexts (one with keystore to support client cert sign and truststore, second just with truststore)
+
+
+        return new NettyRoutingFilterApiml(httpClientNoCert(httpClient), httpClientClientCert(httpClient), headersFiltersProvider, properties);
+    }
+
+    public HttpClient httpClientNoCert(HttpClient httpClient) {
         SslContext justTruststore = sslContext(false);
-        SslContext withKeystore = sslContext(true);
 
         var builderJustTruststore = SslProvider.builder().sslContext(justTruststore);
-        var builderWithKeystore = SslProvider.builder().sslContext(withKeystore);
         if (!nonStrictVerifySslCertificatesOfServices) {
             builderJustTruststore.handlerConfigurator(HttpClientSecurityUtils.HOSTNAME_VERIFICATION_CONFIGURER);
-            builderWithKeystore.handlerConfigurator(HttpClientSecurityUtils.HOSTNAME_VERIFICATION_CONFIGURER);
         }
 
-        // construct http clients with different SSL configuration - with / without client certs
-        var httpClientNoCert = httpClient.secure(builderJustTruststore.build());
-        var httpClientClientCert = httpClient.secure(builderWithKeystore.build());
+        return httpClient.secure(builderJustTruststore.build());
+    }
+
+    public HttpClient httpClientClientCert(HttpClient httpClient) {
```

Comment:
This method seems to be identical to httpClientNoCert, just the input (boolean) is different. I would rather merge them into one method.